### DISCUSSION TITLE
Difference between test and prod when implementing Web3Modal

### DISCUSSION THREAD

**Discussion by GrahamMcBain** [WalletConnect/walletconnect-monorepo#1570] [Developer Support]
Heya. I was testing in a sample project, to see if integrating / switching to web3 modal would solve all my issues (bugs, my fault with other stuff, unreleated), whilst not breaking my current stuff. The test went well so I have begun implementing it in the main project.

With seemingly identical implementations, I get different behaviours however.

In the test project, everything works as expected. However when implemented in my main project, where there shouldn't really be any difference, it is making a call to /wallets as part of getting the preview wallets, on page load.

This is only happening on modal opening in the test project. The reason this is causing an issue, is that when it is called on page load, it is being called before the project ID has been provided in config, so the request URL is missing the project ID, and hence it gets a 400 error and doesn't get the data it needs. Hence, the scrolling preview wallet icons are missing. 
Thats the URL its trying to hit, because project Id isn't provided, as well as the call stack (after minification, sorry). So can tell its trying to preload data, but doing it before it has the project ID, and the sample app I have doesn't seem to be doing that at all...

![log3](https://user-images.githubusercontent.com/36456059/199091141-ae998202-7449-4540-b92f-60dd12134ac1.png)
![log](https://user-images.githubusercontent.com/36456059/199091168-451c7144-58c7-4320-b663-3adc5dfc8762.png)

I can get the preview wallets by calling ExplorerCtrl.getPreviewWallets() just after I have set the confid and provided the project ID, but it shouldn't be doing this itself before the project ID is shown in the first place, and I don't know what could cause that to happen...
The implementation is essentially identical to the first 3 steps from here: https://github.com/WalletConnect/web3modal/blob/V2/docs/html.md

**Comment by pingu2k4** [comment]
I fixed this by removing the <w3m-modal> tag from my index.html, and putting it inside one of my base layouts instead, so that it isn't on the page for the first moments whilst the page loads.
I'm working in Blazor, so not something I expect to have a massive amount of experience within the team with etc, and I don't know react etc well enough to give a comparison, really... but rather than putting the <w3m-modal> tab in index.html, its within a component, and I think thats kind of similar to how react would work as well. When its on the index.html page, there appears to be a race condition (as a previous, for all reasonable purposes identical, project didn't have the same issue)

**Reply by GrahamMcBain** [reply]
Thank you for following up and sharing how you solved this!

**Comment by xzilja** [comment]
Looking into this. Ty for details @pingu2k4 👍

**Comment by NovaBloq** [comment]
I have a similar issue.
I did tried to add the element later on the page, but it didn't helped.

**Comment by xzilja** [comment]
Found the cause, this issue will be solved in next (`beta.2`) release

**Reply by NovaBloq** [reply]
Any eta when it will be released? 
I literally can't do anything because of this.

**Reply by xzilja** [reply]
Targeting end of this week

**Comment by xzilja** [comment]
Beta.2 was released 👍 with fixes for this
